temporal-sdk/src/main/java/io/temporal/internal/sync/DeterministicRunner.java

```diff
@@ -32,8 +32,14 @@
  */
 interface DeterministicRunner {
 
+  boolean debugMode = Boolean.parseBoolean(System.getenv("TEMPORAL_DEBUG_MODE"));
```

Comment:
I would make it static. Getting the variable and parsing it on every workflow sounds excessive. But making it static would require some synchronization.

Reply:
It's defined in the interface, so it's final static by default.